src/components/AnnotationViewer.vue

```diff
@@ -292,6 +311,82 @@ export default class AnnotationViewer extends Vue {
       this.drawNewConnections(displayedIds);
     }
     this.annotationLayer.draw();
+    this.drawTooltips();
+  }
+
+  drawTooltips() {
+    const oldFeatures = this.textLayer.features();
+
+    if (this.shouldDrawTooltips) {
+      // Avoid checking the store for each annotation
+      const all = this.tooltipOnAll;
+      const selected = this.tooltipOnSelected;
+      const hovered = this.tooltipOnHovered;
+
+      const displayedAnnotations = this.annotationLayer
+        .annotations()
+        .map((a: any) => a.options("storedAnnotation"))
+        .filter((a: any) => {
+          return (
+            a &&
+            (all ||
+              (hovered && this.hoveredAnnotationId === a.id) ||
+              (selected && this.isAnnotationSelected(a.id)))
+          );
+        });
+
+      // One text feature per line as in https://opengeoscience.github.io/geojs/tutorials/text/
+      // Centroid is computed once per new line (optimization needed?)
+      // TODO: More performance with renderThreshold https://opengeoscience.github.io/geojs/apidocs/geo.textFeature.html#.styleSpec
+      const baseStyle = {
+        fontSize: "12px",
+        fontFamily: "sans-serif",
+        textAlign: "center",
+        textBaseline: "middle",
+        color: "black",
+        textStrokeColor: "#FFF8",
+        textStrokeWidth: 2
+      };
+      this.textLayer
+        .createFeature("text")
+        .data(displayedAnnotations)
+        .position((annotation: IAnnotation) => {
+          return simpleCentroid(annotation.coordinates);
+        })
+        .style({
+          text: (annotation: IAnnotation) => {
+            return (
+              "XY, Z, T = " +
+              annotation.location.XY +
+              ", " +
+              annotation.location.Z +
+              ", " +
+              annotation.location.Time
+            );
+          },
+          offset: { x: 0, y: -6 },
+          ...baseStyle
+        });
+      this.textLayer
+        .createFeature("text")
+        .data(displayedAnnotations)
+        .position((annotation: IAnnotation) => {
+          return simpleCentroid(annotation.coordinates);
+        })
+        .style({
+          text: (annotation: IAnnotation) => {
+            return "[ " + annotation.tags.join(", ") + " ]";
+          },
+          offset: { x: 0, y: 6 },
+          ...baseStyle
+        });
```

Comment:
Can't you add both texts at the same time?

Reply:
No, that is impossible for now, see comment line 338 `// One text feature per line as in https://opengeoscience.github.io/geojs/tutorials/text/`.